pkg/election/leadership.go

```diff
@@ -213,19 +260,15 @@ func (ls *Leadership) Watch(serverCtx context.Context, revision int64) {
 			for _, ev := range wresp.Events {
 				if ev.Type == mvccpb.DELETE {
 					log.Info("current leadership is deleted",
+						zap.Int64("revision", wresp.Header.Revision),
 						zap.String("leader-key", ls.leaderKey),
 						zap.String("purpose", ls.purpose))
 					return
 				}
 			}
+			revision = wresp.Header.Revision + 1
 		}
-
-		select {
-		case <-ctx.Done():
-			// server closed, return
-			return
-		default:
-		}
+		watchChanCancel()
```

Comment:
If I'm understanding is right, we cancel the channel here immediately after we process line 242 ~ 269, which mightn't be as expected. Shouldn't it goes back and executes select at 233 after line 268?

Reply:
it is copied from loopwatcher, where there is a similar problem. I will also fix it.